lib/_stream_writable.js

```diff
@@ -783,45 +734,81 @@ function onFinished(stream, state, cb) {
   stream.prependListener('error', onerror);
 }
 
-ObjectDefineProperty(Writable.prototype, 'destroyed', {
-  // Making it explicit this property is not enumerable
+ObjectDefineProperties(Writable.prototype, {
+
+  // Making it explicit these below properties are not enumerable
   // because otherwise some prototype manipulation in
   // userland will fail
-  enumerable: false,
-  get() {
-    if (this._writableState === undefined) {
-      return false;
+  destroyed: {
+    enumerable: false,
```

Comment:
I think `ObjectDefineProperties` defaults `enumerable` to false, so I don't think these are needed? I'm not sure I understand the above comment about making it explicit?

Reply:
@ronag: That's a good point. I guess we no need to add these properties. For example, I tested the below code and by default the enumerable property is `false`:

```
const a = {}
Object.defineProperties(a, { test: { get() { return 'test'} } })
a.test // test
a.propertyIsEnumerable('test') // false
```

Reply:
Seems like that was Calvin. Some people prefer explicit descriptor properties. Probably because not everyone is handling descriptors frequently and knows that their defaults are all `false`. It should be fine either way.

Reply:
As this also came up in https://github.com/nodejs/node/pull/31287#discussion_r370203713:

I think it’s better to be explicit here, regardless of what people may know about the default values; partly because it removes cognitive overhead of figuring out what the behaviour is, partly because making it explicit forces code authors to be deliberate about the values they choose.